src/include/migraphx/quantization.hpp

```diff
@@ -20,9 +22,30 @@ void quantize(program& prog);
 std::size_t capture_arguments(program& prog,
                               const std::vector<std::string>& ins_names,
                               const std::function<void(std::size_t, std::vector<argument>)>& func);
+
+std::shared_ptr<std::vector<std::pair<float, float>>> capture_arguments_impl(
+    program& prog, const target& t, const std::vector<std::string>& ins_names = {"dot"});
+
+template <class T>
 std::shared_ptr<std::vector<std::pair<float, float>>>
-capture_arguments(program& prog, const std::vector<std::string>& ins_names);
-std::shared_ptr<std::vector<std::pair<float, float>>> capture_arguments(program& prog);
+capture_arguments(program& prog, T&& t, const std::vector<std::string>& ins_names = {"dot"})
+{
+    static_assert(std::is_same<std::remove_cv_t<std::remove_reference_t<T>>, target>{} &&
+                      std::is_lvalue_reference<T>{},
+                  "Dangling reference to target!");
+    return capture_arguments_impl(prog, t, ins_names);
+}
+
+void quantize_int8(program& prog,
+                   const target& t,
+                   std::vector<program::parameter_map>& calibration_args);
+void quantize_int8(program& prog,
+                   const target& t,
+                   std::vector<program::parameter_map>& calibration_args,
+                   const std::vector<std::string>& ins_names);
```

Comment:
Can you use a default parameter for `ins_names`?

Reply:
fixed.